core/field_dropdown.js

```diff
@@ -184,6 +191,8 @@ Blockly.FieldDropdown.prototype.showEditor_ = function() {
   // Otherwise it will cause a page scroll to get the misplaced menu in
   // view. See issue #1329.
   this.menu_.setAllowAutoFocus(true);
+  // Scrolls HTML div to the selected item.
+  this.menu_.getElement().scroll(0,Blockly.Field.SELECTED_ITEM_Y_POS);
   this.menu_.getElement().focus();
```

Comment:
This won't work on IE, Edge and a couple of other browsers: https://caniuse.com/#search=scroll-behavior
but there's already a utility for scrolling: 
``Blockly.utils.style.scrollIntoContainerView``